pallets/manta-pay/src/lib.rs

```diff
@@ -465,18 +464,38 @@ pub mod pallet {
         T: Config,
     {
         /// Maximum Number of Updates per Shard
-        const PULL_MAX_PER_SHARD_UPDATE_SIZE: usize = 128;
+        const PULL_MAX_RECEIVER_UPDATE_SIZE: u64 = 32768;
 
         /// Maximum Size of Sender Data Update
-        const PULL_MAX_SENDER_UPDATE_SIZE: usize = 1024;
+        const PULL_MAX_SENDER_UPDATE_SIZE: u64 = 32768;
 
         /// Pulls receiver data from the ledger starting at the `receiver_index`.
         #[inline]
-        fn pull_receivers(receiver_index: [usize; 256]) -> (bool, ReceiverChunk) {
+        fn pull_receivers(
+            receiver_index: [usize; 256],
+            max_update_request: u64,
+        ) -> (bool, ReceiverChunk) {
             let mut more_receivers = false;
             let mut receivers = Vec::new();
+            let mut receivers_pulled: u64 = 0;
+            let max_update = if max_update_request > Self::PULL_MAX_RECEIVER_UPDATE_SIZE {
+                Self::PULL_MAX_RECEIVER_UPDATE_SIZE
+            } else {
+                max_update_request
+            };
+
             for (i, index) in receiver_index.into_iter().enumerate() {
-                more_receivers |= Self::pull_receivers_for_shard(i as u8, index, &mut receivers);
+                more_receivers |= Self::pull_receivers_for_shard(
+                    i as u8,
+                    index,
```

Comment:
nit: can we rename `i` and `index` to something explicit like `shard_index` + `utxo/receiver_index`